src/sections/tickets.tsx

```diff
@@ -0,0 +1,149 @@
+          apiToken: loginData.apiToken,
+        })
+      );
+      dispatch(
+        actionsAsync.getTicketsThunk({
+          apiEndpoint: loginData.apiEndpoint,
+          apiToken: loginData.apiToken,
+        })
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loginData]);
+
+  return (
+    <Section className="Section--tickets" id="Section--tickets" yellow>
+      <h2>
+        Tickets{' '}
+        <button
+          onClick={() => {
+            if (loginData.apiEndpoint && loginData.apiToken) {
+              dispatch(
+                actionsAsync.getStatisticsThunk({
+                  apiEndpoint: loginData.apiEndpoint,
+                  apiToken: loginData.apiToken,
+                })
+              );
+              dispatch(
+                actionsAsync.getTicketsThunk({
+                  apiEndpoint: loginData.apiEndpoint,
+                  apiToken: loginData.apiToken,
+                })
+              );
+            }
+          }}
+        >
+          Refresh
+        </button>
+      </h2>
+      <div>Unredeemed: {statistics?.unredeemed}</div>
+      <div>Redeemed: {statistics?.redeemed}</div>
+      <RedeemTickets
+        onSuccess={() => {
+          set_redeemSuccess(true);
+        }}
+        onError={(e) => {
+          set_redeemSuccess(false);
+          set_redeemErrors([
+            ...redeemErrors,
+            {
+              error: e.error,
+              status: e.status,
+            },
+          ]);
+        }}
+      />
+      <div>
+        <button
+          onClick={() => {
+            set_showStatistics(!showStatistics);
+          }}
+        >
+          {showStatistics ? 'Hide Statistics' : 'Show Statistics'}
+        </button>
+        {showStatistics && (
+          <pre>
+            {statistics
+              ? JSON.stringify(statistics, null, 2)
+              : 'No statistics available'}
+          </pre>
+        )}
+      </div>
+      <div>
+        <button
+          onClick={() => {
+            set_showTickets(!showTickets);
+          }}
+        >
+          {showTickets ? 'Hide Tickets' : 'Show Tickets'}
+        </button>
+        {showTickets && (
+          <pre>
+            {tickets
+              ? JSON.stringify(tickets, null, 4)
+              : 'No tickets available'}
+          </pre>
+        )}
+      </div>
+    </Section>
+  );
+}
+
+function RedeemTickets({
+  onError,
+  onSuccess,
+}: {
+  onError: (e: typeof APIError.prototype) => void;
+  onSuccess: () => void;
+}) {
+  const dispatch = useAppDispatch();
+  const loginData = useAppSelector((selector) => selector.auth.loginData);
+
+  return (
+    <button
+      onClick={() => {
+        if (loginData.apiEndpoint && loginData.apiToken) {
+          dispatch(
+            actionsAsync.redeemTicketsThunk({
+              apiEndpoint: loginData.apiEndpoint,
+              apiToken: loginData.apiToken,
+            })
+          )
```

Comment:
I can see that the function took 4.5min for me.
Add a bool 'updating' status to slice of tickets and make it true on redeem start and false on end.
Show the loader we have on connect page in the tickets page when the  'updating === true'

Reply:
Great idea!